server/src/main/java/io/spine/server/aggregate/Aggregate.java

```diff
@@ -519,4 +502,40 @@ public boolean apply(Message message) {
             return !message.equals(EMPTY);
         }
     }
+
+    private static class RecentHistory {
```

Comment:
Can we benefit from having this class used in `Projection`, for example?

Reply:
I think we can. But I’d like to stay within the bounds of this PR. 

Our goal was to implement this functionality for the aggregate commands.

Placing the Recent History outside of Aggregate to lay the groundwork for adding its support to Process Managers or Projections would require a lot of additional work. We would be exposing Recent History as part public API raising many new questions:
- Is this really how the RecentHistory should look like in this case?
- Maybe we would like to have some `PageableIterator` be returned by `iterator()`?  
- We shouldn’t give an iterator over `Event` instances.
- etc.

We will need to work all of this out, but this doesn’t help to solve the issue at hand.